### PR TITLE
Peer connector threads usage optimizations 

### PR DESCRIPTION
**Pull request by noescape00**
Fixes the bug with high threads usage.

### CODE REVIEW

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 1]
```diff
@@ -205,13 +205,16 @@ public void StartConnectAsync()
...
+
             this.asyncLoop = this.asyncLoopFactory.Run($"{this.GetType().Name}.{nameof(this.ConnectAsync)}", async token =>
             {
-                if (!this.peerAddressManager.Peers.Any())
-                    return;
-
-                if (this.ConnectedPeers.Count >= this.MaxOutboundConnections)
+                if (!this.peerAddressManager.Peers.Any() || (this.ConnectedPeers.Count >= this.MaxOutboundConnections))
+                {
+                    await Task.Delay(2000, this.nodeLifetime.ApplicationStopping);
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin/P2P/PeerConnector.cs]: configure await false

**Comment by D3m0nKingx** [comment]
Running node with this new code for 4 hours now. No excessive amount of threads (varies between 100-300), CPU % around 2-70% on avg with occasional spikes to 200-400%. RAM staying around 500mb. Also the node is in a state of IBD atm, currently at 90% sync.

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs (hunk 1 above)
Comment (noescape00) [reply src/Stratis.Bitcoin/P2P/PeerConnector.cs]: Fixed

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 2]
```diff
@@ -205,13 +205,16 @@ public void StartConnectAsync()
 
             this.OnStartConnect();
 
+            if (this.MaxOutboundConnections == 0)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerConnector.cs]: @noescape00 The issue putting the MaxOutboundConnections outside the async loop is that we can add nodes to PeerConnectorAddNode's collection via an RPC call. So if we do this, this connector never starts up and as such when we call that RPC method, it won't have any affect. Remember that PeerConnectorAddNode must always run.

Comment (noescape00) [reply src/Stratis.Bitcoin/P2P/PeerConnector.cs]: Good notice. Didn't know that we can use addNode during the runtime. 
Fixed.

**Comment by fassadlr** [comment]
@noescape00 @D3m0nKingx Are you guys happy with this now? If so I'll merge 👍

**Comment by noescape00** [comment]
@fassadlr sure, got for it.

**Pull request merged**